packages/decoder/lib/decoders.ts

```diff
@@ -742,15 +763,23 @@ export class WireDecoder {
   /**
    * @protected
    */
-  public getWeb3(): Web3 {
-    return this.web3;
+  public getDeployedContexts(): Contexts.Contexts {
+    return this.deployedContexts;
   }
 
   /**
    * @protected
    */
-  public getDeployedContexts(): Contexts.Contexts {
-    return this.deployedContexts;
+  public async send(options: { method: string; params: any[] }): Promise<any> {
+    const { method, params } = options;
+    const { result } = await promisify(this.provider.send.bind(this.provider))({
```

Comment:
I would recommend you check if the provider has a `request` method and use it first. The `send` method [is currently deprecated](https://github.com/MetaMask/providers/blob/e3353443fa4b967ff768d666be1f8937b8d1bfde/src/MetaMaskInpageProvider.ts#L245-L254).